### DISCUSSION TITLE
You Asked, We Answered - New Code Search and View (Beta) FAQs

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#42470] [Other Feature Feedback, Questions, & Ideas]
During Universe 🛰️  2022, our product team dramatically upgraded the code search 🔍  and browsing 📄  experience with the all-new Code Search and Code View beta. Since then, we have received so many great feature requests and questions in the [Beta discussion](https://github.com/orgs/community/discussions/38692). GitHub Community believes in empowering 💪  users to stay on top of their code navigation and search experience and we’d love ❤️  to answer some of your frequently asked questions below:

Some questions are best answered by [this blog post](https://github.blog/changelog/2022-11-09-introducing-an-all-new-code-search-and-code-browsing-experience/) and our [Code Search FAQs ](https://cs.github.com/about/faq#whats-indexed), others we’ll answer below!. 

## How can I request access to the Code Search and View beta program? 🤝 
 
You can sign up and participate in the private beta by joining the [Code Search and Code View Waitlist](https://github.com/features/code-search-code-view/signup). You will receive an email once you are accepted from the waitlist to join the beta. Once given access ✅, the new code search and code view beta will be automatically enabled on your account. 
 
## How do I disable the new Code Search and Code View (beta)? ❔ 
 You can disable the beta on GitHub.com anytime.  
- In the upper-right corner of any page, click your profile photo, then click Feature preview.
- To the right of "New Code Search and Code View (Beta)", click Enable or Disable.

**Note** 📔 : You have to be in the beta program before you can use the feature preview menu. Also, this setting applies to both the code search and the code view experience.
 
## Where do I request a new feature or share any feedback about the new Code Search and View? 🗳️ 

We request you post all feedback and feature requests to this [discussion](https://github.com/orgs/community/discussions/38692https://github.com/orgs/community/discussions/38692) so our product team can track feedback all in one place. 

## Code Search is not working for me and it is returning an error. What should I do? 🔧 

- First, please check that there is no service downtime. To do so, verify the [status](https://www.githubstatus.com/) on our website.
- Whether you’re [searching](https://docs.github.com/en/search-github/github-code-search/about-github-code-search) or [navigating](https://docs.github.com/en/repositories/working-with-files/managing-files/navigating-files-with-the-new-code-view) code, check the GitHub Docs to ensure you’re using it correctly. 
- Don’t forget, the community is full of useful discussions and solutions, refer to the [feedback discussion](https://cs.github.com/about/faq#:~:text=Check%20out%20the-,feedback%20discussion,-to%20see%20if) to see if it’s a known issue and open a new discussion if not.

## How are search results ranked on Code Search? ❓ 

We use a variety of heuristics, including how many matches are in the file, the quality of the matches, the kind of file, whether the searches match symbols, and the number of stars that the repository has, among other factors.

## How can I report a security vulnerability? 🔐 
 
Please use the [GitHub Bug Bounty program](https://bounty.github.com/) to let us know what you found.

## Useful Resources 🔖 

- [GitHub Code Search Docs](https://docs.github.com/en/search-github/searching-on-github/finding-files-on-github)
- [GitHub Code View Docs](https://docs.github.com/en/repositories/working-with-files/managing-files/navigating-files-with-the-new-code-view)
- [Code Search FAQs](https://cs.github.com/about/faq#whats-indexed)
- [New Code Search blog post](https://github.blog/2021-12-08-improving-github-code-search/)
- [New Code Search and View changelog](https://github.blog/changelog/2022-11-09-introducing-an-all-new-code-search-and-code-browsing-experience/)

Learn more about [Code Search](https://github.com/features/code-search) and [Code Navigation](https://github.blog/2022-04-29-bringing-code-navigation-to-communities/) 🚀

**Comment by tripleee** [comment]
[The main announcement](https://docs.github.com/en/repositories/working-with-files/managing-files/navigating-files-with-the-new-code-view) simply links to the [forums](https://github.com/orgs/community/discussions/categories/repositories), could you link it specifically to [the thread where you want the feedback?](https://github.com/orgs/community/discussions/48301)
